src/twisted/conch/ssh/transport.py

```diff
@@ -516,6 +515,99 @@ class SSHTransportBase(protocol.Protocol):
     _keyExchangeState = _KEY_EXCHANGE_NONE
     _blockedByKeyExchange = None
 
+
+    def setSupportedKexes(self, kexes):
+        """
+        Clear out the list of supported key exchanges
+        Then call addSupportedKexes to populate the list.
+
+        @type kexes: L{list}
+        @param kexes: A L{list} of kex name strings
+        """
+        # Clear out the currently supported key exchanges.
+        _kex._kexAlgorithms.clear()
+
+        # Set the new exchanges.
+        self.addSupportedKexes(kexes)
+
+
+    def addSupportedKexes(self, kexes):
```

Comment:
The subtlety is how the list is ordered. :)

Before Twisted ordered the kexes based on their preference ([EC screwed that up a little](http://twistedmatrix.com/trac/ticket/9155)), and to keep doing that I recreate the kex algorithms dictionary and modified `_kex.getSupportedKeyExchanges()`.

But you're right, it shouldn't change things at a module level, so I've made changes to this part of the code so `kexAlgorithms` is now a SSH*Transport class variable.